src/iexamine.cpp

```diff
@@ -640,12 +644,12 @@ void iexamine::rubble(player &p, const tripoint &examp)
         return;
     }
 
-    // "Remove"
-    p.moves -= 200;
-    g->m.furn_set(examp, f_null);
+    // Select our best shovel
+    auto it = std::max_element( shovels.begin(), shovels.end(), []( const item *lhs, const item *rhs ) {
+        return lhs->get_quality( quality_dig ) > rhs->get_quality( quality_dig );
+    } );
```

Comment:
The sorting order seems to be inverted - carrying a stone shovel slows down digging immensely.